src/ProtocolZoo/ProtocolZoo.jl

```diff
@@ -435,6 +470,295 @@ end
+    retry_lock_time::LT = 0.1
+    """how many rounds of this protocol to run (`-1` for infinite))"""
+    rounds::Int = -1
+end
+
+#TODO "convenience constructor for the missing things and finish this docstring"
+function FusionProt(sim::Simulation, net::RegisterNet, node::Int; kwargs...)
+    return FusionProt(;sim, net, node, kwargs...)
+end
+
+@resumable function (prot::FusionProt)()
+    rounds = prot.rounds
+    round = 1
+    while rounds != 0
+        fusable_qubit, piecemaker = findfusablequbit(prot.net, prot.node, prot.nodeC) # request client slots on switch node
+        if isnothing(fusable_qubit)
+            isnothing(prot.retry_lock_time) && error("We do not yet support waiting on register to make qubits available") # TODO
+            @yield timeout(prot.sim, prot.retry_lock_time)
+            continue
+        end
+
+        (q, id, tag) = fusable_qubit.slot, fusable_qubit.id, fusable_qubit.tag
+        (q_pm, id_pm, tag_pm) = piecemaker.slot, piecemaker.id, piecemaker.tag
+        #@yield lock(q) & lock(q_pm) # this should not really need a yield thanks to `findswapablequbits`, but it is better to be defensive
+        #@yield timeout(prot.sim, prot.local_busy_time)
+
+        untag!(q, id)
+        # store a history of whom we were entangled to for both client slot and piecemaker
+        tag!(q, EntanglementHistory, tag[2], tag[3], prot.node, q_pm.idx, q.idx)
+        tag!(q_pm, FusionCounterpart, tag[2], tag[3])
+
+        uptotime!((q, q_pm), now(prot.sim))
+        fuscircuit = EntanglementFusion()
+        zmeas = fuscircuit(q, q_pm) 
+        # send from here to client node
+        # tag with EntanglementUpdateX past_local_node, past_local_slot_idx, past_remote_slot_idx, new_remote_node, new_remote_slot, correction
+        msg = Tag(EntanglementUpdateZ, prot.node, q.idx, tag[3], prot.node, q_pm.idx, zmeas)
+        put!(channel(prot.net, prot.node=>tag[2]; permit_forward=true), msg)
+        @debug "FusionProt @$(prot.node)|round $(round): Send message to $(tag[2]) | message=`$msg`"
+        unlock(q)
+        unlock(q_pm)
+        rounds==-1 || (rounds -= 1)
+        round += 1
+    end
+end
+
+function findfusablequbit(net, node, pred_client)
+    reg = net[node]
+    nodes  = queryall(reg, EntanglementCounterpart, pred_client, ❓; locked=false)
+    piecemaker = query(reg, Piecemaker, ❓, ❓)
+    isempty(nodes) && return nothing
+    @assert length(nodes) == 1 "Client seems to be entangled multiple times"
+    return nodes[1], piecemaker
+end
+
+"""
+$TYPEDEF
+
+A protocol that generates entanglement between two nodes.
+Whenever the selected client slot and the associated slot on the remote node are free, the protocol locks them
+and starts probabilistic attempts to establish entanglement.
+
+$TYPEDFIELDS
+"""
+@kwdef struct SelectedEntanglerProt{LT} <: AbstractProtocol where {LT<:Union{Float64,Nothing}}
```

Comment:
Integrated `slotA` and slotB` variables into `EntanglerProt` which is now serving the need to choose specific register slots at both nodes to be entangled. Since this was the only purpose of the `Selected EntanglerProt` I deleted it.